File: winter-compiler/src/test/resources/WithInjectedProviderAndLazyFields_WinterMembersInjector.java
package test;

import io.jentz.winter.ClassTypeKey;
import io.jentz.winter.GenericClassTypeKey;
import io.jentz.winter.Graph;
import io.jentz.winter.inject.MembersInjector;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.List;
import javax.annotation.Generated;
import kotlin.LazyKt;

@Generated(
        value = "io.jentz.winter.compiler.WinterProcessor",
        date = "2019-02-10T14:52Z"
)
public final class WithInjectedProviderAndLazyFields_WinterMembersInjector implements MembersInjector<WithInjectedProviderAndLazyFields> {
    @Override
    public void inject(final Graph graph, final WithInjectedProviderAndLazyFields target) {
        target.field0 = graph.instanceOrNullByKey(new ClassTypeKey<Object>(Object.class, null));
        target.field1 = () -> graph.instanceOrNullByKey(new GenericClassTypeKey<List<String>>("stringList") {});
        target.field2 = LazyKt.lazy(() -> graph.instanceOrNullByKey(new GenericClassTypeKey<List<String>>("stringList") {}));
    }
}

File: winter-compiler/src/test/resources/WithInjectedProviderAndLazyFields_WinterMembersInjector.kt
package test

import io.jentz.winter.Graph
import io.jentz.winter.inject.MembersInjector
import javax.annotation.Generated
import kotlin.Any
import kotlin.String
import kotlin.collections.List

@Generated(
  value = ["io.jentz.winter.compiler.WinterProcessor"],
  date = "2019-02-10T14:52Z"
)
class WithInjectedProviderAndLazyFields_WinterMembersInjector :
    MembersInjector<WithInjectedProviderAndLazyFields> {
  override fun inject(graph: Graph, target: WithInjectedProviderAndLazyFields) {
    target.field0 = graph.instanceOrNull<Any>()
    target.field1 = { graph.instanceOrNull<List<String>>(qualifier = "stringList", generics = true)
        }
    target.field2 = graph.providerOrNull<List<String>>(qualifier = "stringList", generics = true)
    target.field3 = lazy { graph.instanceOrNull<List<String>>(qualifier = "stringList", generics =
        true) }
  }
}
